src/utils/utils.js

```diff
@@ -1240,15 +1240,25 @@ export function insertText(sel, content) {
 
 /**
  * Add a BR in the given node if its closest ancestor block has nothing to make
- * it visible.
+ * it visible, and/or add a zero-width space in the given node if it's an empty
+ * inline unremovable so the cursor can stay in it.
  *
  * @param {HTMLElement} el
  */
 export function fillEmpty(el) {
+    const fillers = [undefined, undefined];
```

Comment:
why the array with two undefined element ?
An element cannot be ShrunkBlock and EmtpyInline at the same time. Or can it ?

Reply:
Just a way to make explicit that it will _always_ return a tuple.
The element can't indeed but ShrunkBlock is tested on the closest block, not the element. I believe the element can be an empty inline _and_ have a ShrunkBlock as closest block.

Reply:
Seems a bit weird to me to return a fixed size Array, but ok.

Reply:
What would you prefer? An object { br, zws } ?

Reply:
Well it's even better if we can have a thing we both think is good :+1:
I'll do the object thing I think you're right it's better.

Reply:
Done, what do you think?